internal/build/lifecycle_execution.go

```diff
@@ -245,14 +266,25 @@ func (l *LifecycleExecution) Run(ctx context.Context, phaseFactoryCreator PhaseF
 				return err
 			}
 		}
-
+		var start time.Time
 		if l.platformAPI.AtLeast("0.12") && l.hasExtensionsForRun() {
-			group.Go(func() error {
-				l.logger.Info(style.Step("EXTENDING (RUN)"))
-				return l.ExtendRun(ctx, buildCache, phaseFactory)
-			})
+			if l.opts.Publish {
+				group.Go(func() error {
+					l.logger.Info(style.Step("EXTENDING (RUN)"))
+					return l.ExtendRun(ctx, buildCache, phaseFactory)
+				})
+			} else {
+				start = time.Now()
+				group.Go(func() error {
+					l.logger.Info(style.Step("EXTENDING (RUN) BY DAEMON"))
+					defer func() {
+						duration := time.Since(start)
+						fmt.Println("Execution time:", duration)
+					}()
+					return l.ExtendRunByDaemon(ctx, group, &currentRunImage)
+				})
+			}
```

Comment:
```suggestion
			group.Go(func() error {
				l.logger.Info(style.Step("EXTENDING (RUN)"))
				return l.ExtendRun(ctx, buildCache, phaseFactory)
			})
```
(as we saw worse performance using the daemon to extend the run image due to the time spent saving the extended layers out of the daemon)

Reply:
We should add a test for this... or maybe we could just rework the existing tests for `ExtendBuild` to cover the daemon case, since we probably won't use kaniko ever to extend the build image

Reply:
added the tests :)